core/src/rpc_subscriptions.rs

```diff
@@ -913,46 +922,74 @@ impl RpcSubscriptions {
         bank_forks: &Arc<RwLock<BankForks>>,
         commitment_slots: &CommitmentSlots,
         notifier: &RpcNotifier,
+        source: &'static str,
     ) {
+        let mut accounts_time = Measure::start("notify_accounts");
         let pubkeys: Vec<_> = {
             let subs = account_subscriptions.read().unwrap();
             subs.keys().cloned().collect()
         };
+        let mut num_pubkeys_notified = 0;
         for pubkey in &pubkeys {
-            Self::check_account(
+            num_pubkeys_notified += Self::check_account(
                 pubkey,
                 bank_forks,
                 account_subscriptions.clone(),
                 &notifier,
                 &commitment_slots,
-            );
+            )
+            .len();
         }
+        accounts_time.stop();
 
+        let mut programs_time = Measure::start("notify_programs");
         let programs: Vec<_> = {
             let subs = program_subscriptions.read().unwrap();
             subs.keys().cloned().collect()
         };
+        let mut num_programs_notified = 0;
         for program_id in &programs {
-            Self::check_program(
+            num_programs_notified += Self::check_program(
                 program_id,
                 bank_forks,
                 program_subscriptions.clone(),
                 &notifier,
                 &commitment_slots,
-            );
+            )
+            .len();
         }
+        programs_time.stop();
 
+        let mut signatures_time = Measure::start("signatures_programs");
         let signatures: Vec<_> = {
             let subs = signature_subscriptions.read().unwrap();
             subs.keys().cloned().collect()
         };
+        let mut num_signatures_notified = 0;
         for signature in &signatures {
-            Self::check_signature(
+            num_signatures_notified += Self::check_signature(
                 signature,
                 bank_forks,
                 signature_subscriptions.clone(),
                 &notifier,
                 &commitment_slots,
+            )
+            .len();
+        }
+        signatures_time.stop();
+        if (num_pubkeys_notified + num_programs_notified + num_signatures_notified) > 0 {
```

Comment:
It might be nice to know how long it's taking to iterate through the various current subscriptions, even when no notifications are sent, no? Every slot is arguably too often, but might be nice to sample sometimes.

Reply:
how about a time cutoff.. it can log if it takes over 10ms

Reply:
Sounds good